eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -165,185 +162,262 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
+        mongoTemplate.insert(clusteredVariantEntities, ClusteredVariantEntity.class);
+        mongoTemplate.insert(dbsnpClusteredVariantEntities, DbsnpClusteredVariantEntity.class);
 
-        return accessionWrappers;
-    }
+        List<SubmittedVariantOperationEntity> mergeSVOEInsertEntries = new ArrayList<>();
+        List<SubmittedVariantOperationEntity> rsSplitSVOEInsertEntries = new ArrayList<>();
 
-    private List<ClusteredVariantOperationEntity> getClusteredVariantOperations(List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> accessionWrappers,
-                                                                                Map<String,SubmittedVariantEntity> submittedVariantEntitiesGeneratingNewRS){
-        List<ClusteredVariantOperationEntity> clusteredVariantOperationEntities = new ArrayList<>();
-        for(GetOrCreateAccessionWrapper<IClusteredVariant,String,Long> accessionWrapper : accessionWrappers) {
-            Long originalClusteredAccession = submittedVariantEntitiesGeneratingNewRS.get(accessionWrapper.getHash()).getClusteredVariantAccession();
-            ClusteredVariantOperationEntity clusteredVariantOperationEntity = new ClusteredVariantOperationEntity();
-            clusteredVariantOperationEntity.fill(EventType.RS_SPLIT,
-                                                 originalClusteredAccession,
-                                                 accessionWrapper.getAccession(),
-                                                 new StringBuilder().append("Due to RS split new accession id ")
-                                                                    .append(accessionWrapper.getAccession())
-                                                                    .append(" created for remapped accession ")
-                                                                    .append(originalClusteredAccession)
-                                                                    .toString(),
-                                                 null);
-            clusteredVariantOperationEntities.add(clusteredVariantOperationEntity);
+        for (Map.Entry<String, SubmittedVariantOperationEntity> entry : mergeSVOE.entrySet()) {
+            SubmittedVariantOperationEntity svoe = entry.getValue();
+            if(Objects.isNull(svoe.getId())){
+                mergeSVOEInsertEntries.add(svoe);
+                continue;
+            }
+            Query querySubmitted = query(where("_id").is(svoe.getId())
+                    .and("eventType").is(EventType.RS_MERGE_CANDIDATES)
+                    .and("accession").is(svoe.getAccession())
+                    .and("reason").is("RS mismatch with " + svoe.getAccession()));
+            Update update = new Update();
+            update.set("inactiveObjects", svoe.getInactiveObjects());
+            mongoTemplate.upsert(querySubmitted, update, SubmittedVariantOperationEntity.class);
```

Comment:
We know for a fact that the entry exists in SVOE. Then let's make it explicit that we are doing an update and not an upsert.